src/app/fsat/operations/operations.component.ts

```diff
@@ -0,0 +1,176 @@
+import { HelpPanelService } from '../help-panel/help-panel.service';
+import { OperationsService } from './operations.service';
+
+@Component({
+  selector: 'app-operations',
+  templateUrl: './operations.component.html',
+  styleUrls: ['./operations.component.css']
+})
+export class OperationsComponent implements OnInit {
+  @Input()
+  fsat: FSAT;  
+  @Input()
+  fsatOperations: FsatOperations;
+  @Input()
+  settings: Settings;
+  @Output('emitSave')
+  emitSave = new EventEmitter<FsatOperations>();
+  @Input()
+  selected: boolean;
+  @Input()
+  inSetup: boolean;
+  @Input()
+  baseline: boolean;
+  @Input()
+  modificationIndex: number;
+
+  @ViewChild('modalBody', { static: false }) public modalBody: ElementRef;
+
+  @ViewChild('formElement', { static: false }) formElement: ElementRef;
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.setOpHoursModalWidth();
+    this.getBodyHeight();
+  }
+
+  warnings: FanOperationsWarnings;
+  
+  formWidth: number;
+  
+  bodyHeight: number;
+  showOperatingHoursModal: boolean = false;
+  userDefinedCompressibilityFactor: boolean = false;
+
+  idString: string;
+
+  operationsForm: FormGroup;
+
+  constructor(private fsatWarningService: FsatWarningService, private compareService: CompareService, private operationsService: OperationsService, private helpPanelService: HelpPanelService, private fsatService: FsatService) { }
+
+  ngOnInit() {
+    if (!this.baseline) {
+      this.idString = 'fsat_modification_' + this.modificationIndex;
+    }
+    else {
+      this.idString = 'fsat_baseline';
+    }
+    this.init();
+    if (!this.selected) {
+      this.disableForm();
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.selected && !changes.selected.firstChange) {
+      if (this.selected) {
+        this.enableForm();
+      } else {
+        this.disableForm();
+      }
+    }
+    if (changes.modificationIndex && !changes.modificationIndex.firstChange) {
+      this.init();
+    }
+  }
+
+
+  init() {
+    if (this.fsatOperations) {
+      if (!this.fsatOperations.cost) {
+        this.fsatOperations.cost = this.settings.electricityCost;
+      }
+      this.operationsForm = this.operationsService.getFormFromObj(this.fsatOperations);
+      this.save();
+    }
+  }
+
+  disableForm() {
+    this.operationsForm.controls.cost.disable();
+    this.operationsForm.controls.operatingHours.disable();
+  }
+
+  enableForm() {
+    this.operationsForm.controls.cost.enable();
+    this.operationsForm.controls.operatingHours.enable();
+  }
+
+  save(){  
+    this.fsatOperations = this.operationsService.getObjFromForm(this.operationsForm);
+    this.emitSave.emit(this.fsatOperations);
+    this.checkForWarnings();
+  }
+
+  checkForWarnings() {
+    this.warnings = this.fsatWarningService.checkOperationsWarnings(this.fsat, this.settings, !this.baseline);
+  }
+
+  focusField(inputName: string) {
+    if (!this.baseline && inputName === 'measuredVoltage') {
```

Comment:
measuredVoltage will never be the input name here. Think this is copy/paste leftover